int16 step_num
---
string random_str
